src/Command/Email/ConfigurePlatformEmailCommand.php

```diff
@@ -0,0 +1,279 @@
+  }
+
+  /**
+   * @param \AcquiaCloudApi\Connector\Client $client
+   * @param SubscriptionResponse $subscription
+   * @param string $domain_uuid
+   */
+  protected function createDnsText(Client $client, $subscription, $domain_uuid, $file_format): void {
+    $domain_registration_response = $client->request('get', "/subscriptions/{$subscription->uuid}/domains/{$domain_uuid}");
+    $records = [];
+    $this->localMachineHelper->getFilesystem()->remove('dns-records.txt');
+    if ($file_format === 'JSON') {
+      foreach ($domain_registration_response->dns_records as $record) {
+        unset($record->health);
+        $records[] = $record;
+      }
+      $this->logger->debug(json_encode($records));
+      $this->localMachineHelper->getFilesystem()
+          ->dumpFile('dns-records.txt', json_encode($records, JSON_PRETTY_PRINT));
+    }
+    else {
+      foreach ($domain_registration_response->dns_records as $record) {
+        unset($record->health);
+        $records[] = ['type' => $record->type, 'name' => $record->name, 'value' => $record->value];
+      }
+      $this->logger->debug(json_encode($records));
+      $this->localMachineHelper->getFilesystem()
+          ->dumpFile('dns-records.txt', Yaml::dump($records));
+    }
+
+  }
+
+  /**
+   * Polls the Cloud Platform until the registered domain is verified
+   * environment.
+   *
+   * @param \AcquiaCloudApi\Response\SubscriptionResponse $subscription
+   * @param string $domain_uuid
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   */
+  protected function pollDomainRegistrationsUntilSuccess(
+    SubscriptionResponse $subscription,
+    string $domain_uuid,
+    OutputInterface $output
+  ) {
+    // Create a loop to periodically poll the Cloud Platform.
+    $loop = Loop::get();
+    $spinner = LoopHelper::addSpinnerToLoop($loop, 'Waiting for the domain to be verified...', $output);
+    $client = $this->cloudApiClientService->getClient();
+    $this->domain_verification_error = NULL;
+
+    // Poll Cloud every 30 seconds.
+    $loop->addPeriodicTimer(2, function () use ($output, $loop, $client, $subscription, $domain_uuid, $spinner) {
+      try {
+        $response = $client->request('get', "/subscriptions/{$subscription->uuid}/domains/{$domain_uuid}");
+        if ($response->health->code[0] === "4") {
+          $loop->stop();
+          $this->io->error($response->health->details);
+          $this->domain_verification_error = 1;
+        }
+        else if ($response->health->code === "200") {
+          LoopHelper::finishSpinner($spinner);
+          $loop->stop();
+          $output->writeln("\n<info>Your domain is ready for use!</info>\n");
+        }
+        else {
+          $this->logger->debug(json_encode($response));
+        }
+      } catch (AcquiaCliException $exception) {
+        // Do nothing. Keep waiting and looping and logging.
+        $this->logger->debug($exception->getMessage());
+      }
+    });
+    $loop->run();
```

Comment:
Looks like you removed the timeout... Seems like there isn't really a reason to use a loop anymore. Just use the parent while().

Reply:
Could the periodicTimer function still be of use here?